lib/pbench/test/unit/server/query_apis/test_search.py

```diff
@@ -0,0 +1,183 @@
+import pytest
+from http import HTTPStatus
+from pbench.test.unit.server.query_apis.commons import Commons
+
+
+class TestSearch(Commons):
+    """
+    Unit testing for resources/SearchAPI class.
+
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        self.endpoint = "/index/search"
+        self.json = {
+            "user": "drb",
+            "start": "2020-08",
+            "end": "2020-10",
+            "search_term": "random_string",
+        }
+        self.json_bad_date = {
+            "user": "drb",
+            "start": "2020-12",
+            "end": "2020-19",
+            "search_term": "random_string",
+        }
+        self.json_error = {"user": "", "start": "2020-08", "end": "2020-08"}
+        self.empty_response_payload = {
+            "took": 4,
+            "timed_out": "false",
+            "_shards": {"total": 5, "successful": 5, "skipped": 2, "failed": 0},
+            "hits": {
+                "total": {"value": 0, "relation": "eq"},
+                "max_score": "null",
+                "hits": [],
+            },
+        }
```

Comment:
In fact, maybe this *should* be a default, since I think it's pretty much shared by all the `_search` queries?

Reply:
If we arrange for the base class to initialize these attributes itself, then it can apply appropriate defaults unless a subclass requests otherwise.